docs/DeveloperGuide.md

```diff
@@ -1,38 +1,177 @@
 # Developer Guide
 
-## Acknowledgements
+## Design
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{To Be Added}
 
```

Comment:
Can probably add the overview design to help the user get familiar with the project